### PR TITLE
[第9組]Refactor naming convetions

### PR DESCRIPTION
**Pull request by Yi-Cheng-Wang**
# 1. Variable Naming
The modified code changes the original short variable names to more descriptive names, improving code readability. For example:
- The original code used `db` to represent the database connection; it has been changed to `database` or `db_session`, clearly indicating the purpose of the variable.
- The original code used `n` to represent news articles; it has been changed to `news_item` or `article`, making it easier to understand.
- The original code used `u` to represent users; it has been changed to `user`, aligning better with the semantics of the code.

# 2. Constant Extraction
The modified code extracts constant values scattered throughout the original code and defines them as named constants, improving code maintainability. For example:
- Database connection string: `DATABASE_URL`.
- Sentry service DSN: `SENTRY_DSN`.
- JWT secret key: `JWT_SECRET_KEY`.
- Default times for operations: `DEFAULT_TOKEN_EXPIRE_MINUTES`, `FETCH_INTERVAL_MINUTES`.

### CODE REVIEW

**Comment by siangting** [comment]
The comment on the PR is excellent.

**Comment by EdwardKYC** [comment]
整體命名十分清晰，尤其是在處理新聞數據的邏輯上，代碼命名能夠很直觀地表達出其功能。
get_new_info建議改為get_news_info較不容易誤解。建議改為 fetch_news_data 或 retrieve_news_metadata。

**Comment by Yi-Cheng-Wang** [comment]
> 整體命名十分清晰，尤其是在處理新聞數據的邏輯上，代碼命名能夠很直觀地表達出其功能。 get_new_info建議改為get_news_info較不容易誤解。建議改為 fetch_news_data 或 retrieve_news_metadata。

Thank you for your compliment. The reason get_new_info hasn't been changed is that it wolud affect test case.

**Comment by WANG-YongLE** [comment]
建議：
將所有常數統一放置在特定區域，以保持代碼的一致性與可讀性。
刪除 184~185 行中的迴圈。
277 行的改動看不出明確意義

**Comment by Yi-Cheng-Wang** [comment]
> 建議：
> 將所有常數統一放置在特定區域，以保持代碼的一致性與可讀性。
> 刪除 184~185 行中的迴圈。
> 277 行的改動看不出明確意義


Thank you for your suggestions.

First, the variables being scattered in different places does make things a bit messy. If it were up to me, I would prefer to place these variables in settings.py or config.py.

Second, since this change is about modifying variables and not optimizing the code, this part should not be altered.

Third, changing `oauth2_scheme` to all uppercase is because this variable is meant to remain constant, so it’s better to use uppercase.

**Comment by kevin0216** [comment]
Overall, it's a well refactoring attempt, but there are some suggestions:
- oauth2_scheme which utilize class OAuth2PasswordBearer should be a service, which is not a constant. Its functionality should be generate a Bearer Token from OAuth2 flow, which is dynamically generated in every session. So oauth2_scheme should remain lowercased instead of full uppercase in this changes.
- the underscore in _id_counter is telling IDE this variable is a private variable (but it won't affect in Python Interpreter), it should not be removed.
- For some constants, you should put them to the begin of the code, otherwise they seem kind of unnecessary.

File: backend/main.py [hunk 1]
```diff
@@ -22,24 +22,29 @@
 
 Base = declarative_base()
 
+USER_NEWS_ASSOCIATION_TABLE_NAME = "user_news_upvotes"
+USERS_TABLE_NAME = "users"
+NEWS_ARTICLES_TABLE_NAME = "news_articles"
 
 user_news_association_table = Table(
-    "user_news_upvotes",
+    USER_NEWS_ASSOCIATION_TABLE_NAME,
```
Comment (taro0513) [inline backend/main.py:30]: The name is located under the Table. I think it is enough to describe its purpose and significance. If you want to provide more information, just use an explicit way to implement the class

`user_news_association_table = Table(name="user_news_upvotes", ...)`

File: backend/main.py [hunk 2]
```diff
@@ -157,138 +169,141 @@ def get_new_info(search_term, is_initial=False):
     all_news_data = []
     # iterate pages to get more news data, not actually get all news data
     if is_initial:
-        a = []
-        for p in range(1, 10):
-            p2 = {
-                "page": p,
+        news_pages = []
+        MAX_PAGES = 10
```
Comment (taro0513) [inline backend/main.py:173]: Defining constants outside of functions is generally recommended : )

**Review by taro0513 — COMMENTED** [review]
LGTM

**Pull request merged**